decidim-core/lib/decidim/amendable.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This concern contains the logic related to amendable resources.
+  module Amendable
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :amendments, as: :amendable, foreign_key: "decidim_amendable_id", foreign_type: "decidim_amendable_type", class_name: "Decidim::Amendment"
+
+      # resource.emendations : resources that have amend the resource
+      has_many :emendations, through: :amendments, source: :emendation, source_type: self.name, inverse_of: :emendations
+
+      # resource.amenders :  users that have emendations for the resource
+      has_many :amenders, through: :amendments, source: :amender
+
+      # resource.amended : the original resource that was amended
+      has_one :amend, as: :amendable, foreign_key: "decidim_emendation_id", foreign_type: "decidim_emendation_type", class_name: "Decidim::Amendment"
+      has_one :amendable, through: :amend, source: :amendable, source_type: self.name
+
+    end
+
+    def emendation?
```

Comment:
I don't understand the name of this method 😅

Reply:
this method could be considered the opposite of `amendable?` because (for now) emendations are not amendable
So:
 `resource.amendable? ? "I'm the original proposal" : "I'm the emendation"`
and
`resource.emendation? ? "I'm the emendation" : "I'm the original proposal"` 

for instance is used in `AmendmentsHelper`:

https://github.com/decidim/decidim/blob/626eb1ee782f8232aa35c0b289404ef222c47600/decidim-core/app/helpers/decidim/amendments_helper.rb#L30-L32

https://github.com/decidim/decidim/blob/626eb1ee782f8232aa35c0b289404ef222c47600/decidim-core/app/helpers/decidim/amendments_helper.rb#L41-L43

Reply:
another use case https://github.com/decidim/decidim/blob/1c4b650e0af02cb5ab5a788c4ffa47b4bb60477f/decidim-proposals/app/controllers/decidim/proposals/proposals_controller.rb#L41